ddtrace/internal/sampling.py

```diff
@@ -208,44 +210,62 @@ def apply_span_sampling_tags(self, span):
 
 def get_span_sampling_rules():
     # type: () -> List[SpanSamplingRule]
-    json_rules_raw = os.getenv("DD_SPAN_SAMPLING_RULES")
-    if json_rules_raw is None:
-        return []
+    env_json_rules = os.getenv("DD_SPAN_SAMPLING_RULES")
+    file_json_rules = os.getenv("DD_SPAN_SAMPLING_RULES_FILE")
+    if env_json_rules and file_json_rules:
+        log.warning(
+            (
+                "DD_SPAN_SAMPLING_RULES and DD_SPAN_SAMPLING_RULES_FILE detected. "
+                "Defaulting to DD_SPAN_SAMPLING_RULES value."
+            )
+        )
+        raw_json_rules = env_json_rules
+    elif env_json_rules:
+        raw_json_rules = env_json_rules
+    elif file_json_rules:
+        with open(file_json_rules) as f:
+            raw_json_rules = f.read()
+    # No rules specified
     else:
-        sampling_rules = []
+        return []
+
+    try:
+        json_rules = json.loads(raw_json_rules)
+        if not isinstance(json_rules, list):
+            raise TypeError("DD_SPAN_SAMPLING_RULES is not list, got %r" % json_rules)
+    except JSONDecodeError:
+        raise ValueError("Unable to parse DD_SPAN_SAMPLING_RULES=%r" % env_json_rules)
+
+    sampling_rules = []
+    for rule in json_rules:
+        if not isinstance(rule, dict):
+            raise TypeError("rule specified via DD_SPAN_SAMPLING_RULES is not a dictionary:%r" % rule)
+        # If sample_rate not specified default to 100%
+        sample_rate = float(rule.get("sample_rate", 1.0))
+        service = rule.get("service")
+        name = rule.get("name")
+        # If max_per_second not specified default to no limit
+        max_per_second = int(rule.get("max_per_second", -1))
+        if not isinstance(service, (string_types, type(None))):
```

Comment:
```suggestion
        if service is not None and not isinstance(service, string_types):
```

?

Reply:
Ok I'll change it, but what's the difference between these two?